### PR TITLE
Added OpenInternPositionDTO class + refactored filtrations

### PR DESCRIPTION
**Pull request by meOtniel**

### CODE REVIEW

File: src/main/java/com/sintern/api/OpenInternPositionController.java [hunk 1]
```diff
@@ -12,22 +13,22 @@
...
     @Autowired
     public OpenInternPositionController(OpenInternPositionService openInternPositionService) {
         this.openInternPositionService = openInternPositionService;
     }
-
-    @RequestMapping(value = "/FilterByDomain", method = RequestMethod.GET)
-    public List<OpenInternPosition> filterByDomain(DomainType domainType){
-        return openInternPositionService.findOpenInternPositionByCompany_Domain(domainType);
+    
+    @RequestMapping(value = "/domain", method = RequestMethod.GET)
```
Comment (DascaluCosmin) [inline src/main/java/com/sintern/api/OpenInternPositionController.java]: The API should be **"/open-positions/domain/{domainType}"**, where **domainType** is a Path Variable (**@PathVariable**). Thus, change the value to = **"/domain/{domainType}"** and annotate **DomainType domainType** with **@PathVariable**.

File: src/main/java/com/sintern/api/OpenInternPositionController.java [hunk 2]
```diff
@@ -12,22 +13,22 @@
...
+    public List<OpenInternPositionDTO> filterByDomain(DomainType domainType){
+        return openInternPositionService.findOpenInternPositionDTOByCompany_Domain(domainType);
     }
 
-    @RequestMapping(value = "/FilterByAddress", method = RequestMethod.GET)
-    public List<OpenInternPosition> filterByAddress(String address){
-        return openInternPositionService.findOpenInternPositionByCompany_Address(address);
+    @RequestMapping(value = "/address", method = RequestMethod.GET)
```
Comment (DascaluCosmin) [inline src/main/java/com/sintern/api/OpenInternPositionController.java]: Same as above.

File: src/main/java/com/sintern/service/OpenInternPositionService.java [hunk 3]
```diff
@@ -17,10 +19,27 @@ public OpenInternPositionService(OpenInternPositionRepository openInternPosition
         this.openInternPositionRepository = openInternPositionRepository;
     }
 
-    public List<OpenInternPosition> findOpenInternPositionByCompany_Address(String address){
```
Comment (DascaluCosmin) [inline src/main/java/com/sintern/service/OpenInternPositionService.java:20]: Remove the **_** character.

File: src/main/java/com/sintern/domain/OpenInternPositionDTO.java [hunk 4]
```diff
@@ -0,0 +1,108 @@
+package com.sintern.domain;
+
+import com.fasterxml.jackson.annotation.JsonBackReference;
+
+import javax.persistence.FetchType;
+import javax.persistence.ManyToOne;
+import java.util.UUID;
+
+public class OpenInternPositionDTO {
+    UUID id;
```
Comment (DascaluCosmin) [inline src/main/java/com/sintern/domain/OpenInternPositionDTO.java]: These fields should be private.

File: src/main/java/com/sintern/api/OpenInternPositionController.java [hunk 5]
```diff
@@ -12,22 +13,22 @@
 import java.util.List;
 
 @RestController
-@RequestMapping("/API/OpenInternPosition")
+@RequestMapping("/open-positions")
 public class OpenInternPositionController {
     public OpenInternPositionService openInternPositionService;
```
Comment (DascaluCosmin) [inline src/main/java/com/sintern/api/OpenInternPositionController.java]: Make this **private**.

File: src/main/java/com/sintern/api/OpenInternPositionController.java (hunk 1 above)
Comment (meOtniel) [reply src/main/java/com/sintern/api/OpenInternPositionController.java]: done

File: src/main/java/com/sintern/api/OpenInternPositionController.java (hunk 2 above)
Comment (meOtniel) [reply src/main/java/com/sintern/api/OpenInternPositionController.java]: done

File: src/main/java/com/sintern/service/OpenInternPositionService.java (hunk 3 above)
Comment (meOtniel) [reply src/main/java/com/sintern/service/OpenInternPositionService.java:20]: done

File: src/main/java/com/sintern/domain/OpenInternPositionDTO.java (hunk 4 above)
Comment (meOtniel) [reply src/main/java/com/sintern/domain/OpenInternPositionDTO.java]: done

File: src/main/java/com/sintern/api/OpenInternPositionController.java (hunk 5 above)
Comment (meOtniel) [reply src/main/java/com/sintern/api/OpenInternPositionController.java]: done

**Pull request merged**